### ISSUE TITLE
filter_input on fast-cgi

### ISSUE THREAD

**Issue by mahype** [felixarntz/plugin-lib#4]
I had the problem with Torro Forms that the filter_input function returns null on different fast-cgi PHP installations. For example we need the request URI for the forms:

`filter_input( INPUT_SERVER, 'REQUEST_URI' )`

I found this here:
https://stackoverflow.com/questions/25232975/php-filter-inputinput-server-request-method-returns-null

Also I found it in the comments of the function documentation:
http://php.net/manual/de/function.filter-input.php#77307

This seems to be a PHP which also exists in 7.0 and existing since 11 years ago.

To get sure, to retrieve a value, we should add a wrapper function. Something like:
```php
function my_filter_input( $type, $variable_name, $filter = FILTER_DEFAULT, $options = null ) {
		if ( null === filter_input( $type, $variable_name ) ) {
			$types = array( INPUT_GET, INPUT_POST, INPUT_COOKIE );

			if ( in_array( $type, $types, true ) || filter_has_var( $type, $variable_name ) ) {
				return filter_input( $type, $variable_name, $filter, $options );
			} elseif ( INPUT_SERVER === $type && isset( $_SERVER[ $variable_name ] ) ) {
				return filter_var( wp_unslash( $_SERVER[ $variable_name ] ), $filter, $options );
			} elseif ( INPUT_ENV === $type && isset( $_ENV[ $variable_name ] ) ) {
				return filter_var( $_ENV[ $variable_name ], $filter, $options );
			} else {
				return null;
			}
		}
	}
```

**Comment by felixarntz** [comment]
Wow, that's a bad bug. I'm okay with adding a wrapper function. We just need to think about where to put it best.

About the code itself, some comments:
* The first `filter_input()` call result should be stored in a variable, then checked for `null` afterwards.
* The whole `null` check and if-clause should only run when `$type` is either `INPUT_SERVER` or `INPUT_ENV`, since the other ones are not affected by the bug according to the documentation.
* Instead of using all these `if` / `elseif` statements, it would be cleaner to just have separate `if` statements. By returning anyway it will have the same effect.

**Comment by mahype** [comment]
Yes I just copied this example function from annother site. I will give you a pull request for that.
